src/Adapter/Product/AdminProductWrapper.php

```diff
@@ -212,7 +212,11 @@ public function processQuantityUpdate(Product $product, $quantity, $forAttribute
     {
         // Hook triggered by legacy code below: actionUpdateQuantity('id_product', 'id_product_attribute', 'quantity')
         StockAvailable::setQuantity((int) $product->id, $forAttributeId, $quantity);
-        Hook::exec('actionProductUpdate', ['id_product' => (int) $product->id, 'product' => $product]);
+        static $hookExecuted = [];
```

Comment:
Please put out the method the attribute and define it a description. Thanks

Reply:
sure? the attribute $hookExecuted is just a variable. Its scope is just this method. The use of the definition static here is just for notifying the compiler to recycle the same memory for every execution and it is totally accepted.
Here is explained perfectly: https://www.php.net/manual/en/language.variables.scope.php (search for Example #5 Example use of static variables)
I think that putting out this attribute it will become into garbage for the class, it is totally useless for any other code line.
However, I may be wrong. If you confirm me I will put this attribute outside

Reply:
Hi @prestaquality the idea is quite good however I see one possible issue / limitation: testing.

If we decide tomorrow to write a set of unit tests for this class, this static variable will be shared among them, which will break the necessary isolation property of these unit tests. It could make our tests very hard to maintain.

So indeed having the variable as a class instance will be a bigger memory consumption, but it will make the code a little more maintainable. It seems like a nice compromise 🤔wdyt ? An array of IDs is not something we should be afraid of I think.

Btw if we are talking about optimization, maybe using `array_key_exists()` or `isset()` is faster than `in_array()` ? (so we would transform `$hookExecuted[] = $product->id` into `$hookExecuted[$product->id] = true;`)

Reply:
OMG you are right with the in_array statement... but... before to find a perfect way for this. What if this hook calling were removed? I created this fix because I am not really sure if this calling is here because of any legacy trouble... The hook productUpdate is also call in update() function inside of the Product entity (and also in updateWS). And there is other hook for notifiying when stock is changed fired by StockAvailable... so, may we are discussing here for solve this performance trouble finding the perfect way and it is just removing the line.